File: tools/plan-check.c
#include <stdio.h>

#include <libitute164.h>

int main(int argc, char **argv)
{
    if(argc != 2) {
        fprintf(stderr, "usage: %s <e164-plan.txt>\n", argv[0]);
        return 2;
    }

    if(itu_t_e164_load_plan_file(argv[1]) != 0) {
        const char *error = itu_t_e164_plan_error();
        fprintf(stderr, "%s: invalid numbering plan", argv[1]);
        if(error != NULL)
            fprintf(stderr, ": %s", error);
        fprintf(stderr, "\n");
        return 1;
    }

    return 0;
}

File: js/tools/plan-check.js
#!/usr/bin/env node

import { readFileSync } from "node:fs";
import { basename } from "node:path";
import { E164Plan } from "../src/index.js";

const program = basename(process.argv[1]);
const path = process.argv[2];

if (process.argv.length !== 3) {
  console.error(`usage: ${program} <e164-plan.txt>`);
  process.exit(2);
}

try {
  const text = readFileSync(path, "utf8");
  E164Plan.fromText(text);
} catch (error) {
  console.error(`${path}: invalid numbering plan: ${error.message}`);
  process.exit(1);
}
